x/distribution/keeper/invariants.go

```diff
@@ -1,25 +1,34 @@
-package simulation
+package keeper
 
 import (
 	"fmt"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	distr "github.com/cosmos/cosmos-sdk/x/distribution"
 	"github.com/cosmos/cosmos-sdk/x/distribution/types"
 )
 
+// register all distribution invariants
+func RegisterInvariants(c types.CrisisKeeper, k Keeper, stk types.StakingKeeper) {
+	c.RegisterRoute(types.ModuleName+"/nonnegative-outstanding",
```

Comment:
Can we make `RegisterRoute` to take two routes name and concat them internally?

```go
func (c CrisisKeeper) RegisterRoute(moduleName, invarType string, h InvariantHandler) {
    route := moduleName + "/" + invarType
    //...
}
```

This way we can know about which module has failed and use that information on future circuit breakers.

Reply:
Or, we can leave `RegisterRoute` as it is and modify `type Invariant` into `func (ctx Context, invarType string)` which seems more uniform with the existing router designs.

Reply:
Your latter suggestion really over-complicates things, I really can't see why we would want to include a router as a part `type Invariant`. I'm not sure I understand the problem you're attempting to resolve here. 

As per your first suggestion, yes I agree, If you're looking to pass information as to which module failed, that information can be made available to the circuit breaker _within_ the `crisis.Router` namely within the handler. I will separate out this information within the crisis message and the `RegisterRoute` function

Reply:
addressed in https://github.com/cosmos/cosmos-sdk/pull/3656/commits/a90cecca6c7766a9c7aac0a74956a8d688c2540c